components/controls/ClipboardIconButton.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import * as classNames from 'classnames';
+import Tooltip from './Tooltip';
+import { Button } from './buttons';
+import ClipboardBase from './ClipboardBase';
+import { translate } from '../../helpers/l10n';
+
+interface State {
+  isCopied: boolean;
+}
+
+export default class ClipboardIconButton extends ClipboardBase {
+  copyButton?: HTMLElement | null;
+  mounted = false;
+  state: State = { isCopied: false };
+
+  showTooltip = () => {
+    if (this.mounted) {
+      this.setState({ isCopied: true });
+      setTimeout(() => {
+        if (this.mounted) {
+          this.setState({ isCopied: false });
+        }
+      }, 1000);
+    }
+  };
+
+  render() {
+    const tooltipProps = {
+      overlay: translate(this.state.isCopied ? 'copied_action' : 'copy_to_clipboard'),
+      ...(this.state.isCopied && { visible: true })
+    };
+
+    return (
+      <Tooltip {...tooltipProps}>
+        <Button
```

Comment:
Instead of using tooltip and button and introducing a new kind/style of icon buttons, I think we should use the `ButtonIcon`. It has a tooltip prop too.

Reply:
Ok but currently the `ButtonIcon` tooltip is broken because of the change with the `ThemeContext`, I'll push a small fix to it.